proposals/Integrate with external device inventory.md

```diff
@@ -0,0 +1,44 @@
+# Integrate with external device inventory
+
+## Background
+Akri extracts device information (onvif,opcua or udev device) only from discovery procedure. 
+
+
+For example, Akri discovers onvif camera and extracts information such as video stream source URI through ONVIF discovery and ONVIF command. Video source URI is then used in broker POD to build connection with the camera.
+
+But there are other important device information that can not be extracted through discovery procedure itself. It limits Akri being used in real world production environment.
+
+1.  If ONVIF devices are protected by credential, Akri agent won't be able to find out video stream source URI because the ONVIF command need pass authentication. If RTSP stream is also protected with its own credential, broker POD will not be able to receive video stream. Same problem happens when OPC UA servers (such as facotry PLCs) only serve those OPCUA clients having approved X509 cert and related private key. Broker pod is unable to build a OPC UA connection since Akri does not have PLCs's authentication information.
+
+2. Not only credential, there maybe also important device meta data information that is essential for business logic implementation in broker POD, such as the PLC machine's serial number or asset hierachy information.
+
+3. Real world projects may ask for a proper device provisioning enrollment procedure. Before a device is enrolled, even it is discovered, broker POD should not be created and started. Akri acts on discovery automatically now without considering device provisioning.  
+
+The above issues can be resolved by allowing Akri getting additional device infomration from an external optional device inventory system. This proposal will introduce a few additional steps in Akri's workflow to allow Akri extracting addtioanl device information from external source. 
+
+<br/><br/>
+
+## Proposal
+
+The below workflow presents a new flow from discovering device till creating broker POD. Additional steps are highlighted as yellow color.
+<img src="../media/integrate-extenal-device-inventory-diagram.svg" style="padding-bottom: 10px padding-top: 20px; margin-right: auto; display: block; margin-left: auto;background-color:white"/>
+
+- Introduce a new setting that user can specify the name of a customized external device inventory POD or service, so Akri agent can request for additional device information. The proposed external device inventory POD/service must expose a REST POST API /queryDevice for Akri Agent to raise request.
```

Comment:
Would be good to add a section defining the API in detail, specifically the contents of the query